backend/routes/post_creation.js

```diff
@@ -0,0 +1,41 @@
+const express = require("express");
+const router = express.Router();
+const { createPost } = require("../utils/post.js");
+
+router.post("/create-post", async (req, res) =>
+{
+    const { user_id, song_link, content } = req.body;

```

Comment:
very quick fix: it is best to retrieve the paramaters indivually such as with const userID=req.body.user_id or something similar, this in order to account for a case where the frontend doesn't send the parameters in the expected order (this same thing actually caused issues a few PRs ago), should probably also require a small modification to the unit tests to account for this new way of getting the parameters

Reply:
also if you could add a check for when the parameters are missing in the request that would be great, along with a few tests for it